Vue migration (step1: Sidebar dynamic menus on all system pages)

powpilot opened the pull request:
Replacing sidebar dynamic menus on all system pages

powpilot committed:
added menu.js into the js-library

menu.js fetches dynamic menu atributtes from the server via Vue.j

powpilot committed:
added menu.js into the js-library

menu.js fetches dynamic menu atributtes from the server via Vue.js

powpilot committed:
Delete menu.js from wrong directory

powpilot committed:
dynamic menu on Vue js

replaced tarantool-based dynamic menu with the dynamic menu on Vue js

powpilot committed:
Replaced the sidebar menu with the Vue one

Replaced tarantool based dynamic sidebar menu with the new one (on Vue)

powpilot committed:
added dynamic sidebar menu (on Vue)

Replaced tarantool based dynamic sidebar menu with the new one (on Vue)

powpilot committed:
Replaced the dynamic sidebar menu with Vue one

Replaced tarantool based dynamic sidebar menu with the new one (on Vue)

powpilot committed:
Replaced the dynamic sidebar menu with Vue one

Replaced tarantool based dynamic sidebar menu with the new one (on Vue)

powpilot committed:
Replaced the dynamic sidebar menu with Vue one

Replaced tarantool based dynamic sidebar menu with the new one (on Vue)

vvzvlad commented on templates/system/bus_storage.html:
```diff
@@ -41,14 +41,21 @@
+                                </a>
+                            </li>
+                        </div>
+		                <script src="system-js/menu.js">  
```
Скрипт в теле страницы? Фу.
Скрипты лежат в одном месте, после html-a, в конце страницы(чтобы грузилось быстрее)

vvzvlad commented on public/system-js/menu.js:
```diff
@@ -0,0 +1,21 @@
+},
+	
+        created(){
+fetch('http://192.168.1.45:8080/system_menu_data')
```
А что, если адрес сервера поменяется, все перестанет работать? Заменить на fetch('system_menu_data')

vvzvlad commented on public/system-js/menu.js:
```diff
@@ -0,0 +1,21 @@
+
+
+var dynamenu = new Vue({
```
Код не отформатирован. Поставь какой-нибудь VS Code

vvzvlad commented on templates/system/bus_storage.html:
```diff
@@ -41,14 +41,21 @@
-                        % end
+
+                        
+                     <!-- Beigining of dynamic menu via Vue js -->
```
Beginning?

vvzvlad commented on public/system-js/menu.js:
```diff
@@ -0,0 +1,21 @@
+.then(json=> {
+
+this.menuitems= json;
+	console.log(this.menuitems);
```
Зачем логгирование в продакшене? И вообще, там есть для хрома расширение для просмотра данных в vue.

vvzvlad commented on templates/system/bus_storage.html:
```diff
@@ -24,7 +24,7 @@
     <link href="system-css/dashboard.css" rel="stylesheet">
     <link href="local_settings.css" rel="stylesheet">
-
+    <script src="https://unpkg.com/vue"></script> <!-- Pluging  Vue  in -->
```
Скачай все-таки минифицированный vue и положи его файликом в public. Все, что лежит в public, подключается так же, как "<link href="local_settings.css" rel="stylesheet">" выше, т.е. без указания пути

powpilot commented:
ОК, делаю.
    On Tuesday, May 22, 2018, 2:55:34 PM GMT+3, vvzvlad <notifications@github.com> wrote:  
 
 
@vvzvlad requested changes on this pull request.

In templates/system/bus_storage.html:
> -                                      <%= (v.name or "No name") %>
-                                  </a>
-                        </li>
-                        % end
+
+                        
+                     <!-- Beigining of dynamic menu via Vue js -->
+                        <div id="dynamenu">
+                            <li  v-for="menuitem in menuitems">
+                                <a  class="nav-link" v-bind:href="menuitem.href" > 
+		                        <i v-bind:class="menuitem.icon"></i>
+                                {{menuitem.name}}
+                                </a>
+                            </li>
+                        </div>
+		                <script src="system-js/menu.js">  

Скрипт в теле страницы? Фу.
Скрипты лежат в одном месте, после html-a, в конце страницы(чтобы грузилось быстрее)

In public/system-js/menu.js:
> @@ -0,0 +1,21 @@
+
+
+var dynamenu = new Vue({
+  el: '#dynamenu',
+  data: 
+{    menuitems:  []    
+},
+	
+        created(){
+fetch('http://192.168.1.45:8080/system_menu_data')

А что, если адрес сервера поменяется, все перестанет работать? Заменить на fetch('system_menu_data')

In public/system-js/menu.js:
> @@ -0,0 +1,21 @@
+
+
+var dynamenu = new Vue({

Код не отформатирован. Поставь какой-нибудь VS Code

In templates/system/bus_storage.html:
> @@ -41,14 +41,21 @@
             <nav class="col-md-1 d-none d-md-block bg-light sidebar">
                 <div class="sidebar-sticky">
                     <ul class="nav flex-column">
-                        % for i,v in pairs(menu) do
-                        <li class="nav-item">
-                            <a class="nav-link <%= (v.class or " ")%>" href="<%= v.href or '#' %>">
-                                      <i class="<%= (v.icon or "") %>"></i>
-                                      <%= (v.name or "No name") %>
-                                  </a>
-                        </li>
-                        % end
+
+                        
+                     <!-- Beigining of dynamic menu via Vue js -->

Beginning?

In public/system-js/menu.js:
> @@ -0,0 +1,21 @@
+
+
+var dynamenu = new Vue({
+  el: '#dynamenu',
+  data: 
+{    menuitems:  []    
+},
+	
+        created(){
+fetch('http://192.168.1.45:8080/system_menu_data')
+.then(response => response.json())
+.then(json=> {
+
+this.menuitems= json;
+	console.log(this.menuitems);

Зачем логгирование в продакшене? И вообще, там есть для хрома расширение для просмотра данных в vue.

In templates/system/bus_storage.html:
> @@ -24,7 +24,7 @@
     <!-- Custom styles for this template -->
     <link href="system-css/dashboard.css" rel="stylesheet">
     <link href="local_settings.css" rel="stylesheet">
-
+    <script src="https://unpkg.com/vue"></script> <!-- Pluging  Vue  in -->

Скачай все-таки минифицированный vue и положи его файликом в public. Все, что лежит в public, подключается так же, как "" выше, т.е. без указания пути

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

powpilot committed:
deleted menu.js

deleted the wrong file from wrong path

powpilot committed:
uploaded the corrected menu.js

1.changed endpoint url from absolute to relative
2.removed logging
3.updated the lines formatting (beautified)

powpilot committed:
uploaded minified vue library

powpilot committed:
correction updates following Vlad's comments

1.  moved main.js link to the bottom of html page
2.  corrected comments misspellings 
3.  changed the vue.js plugging from cdn to local and from full-dev to minified.

powpilot committed:
some minor correctrion of a misspelling

powpilot committed:
correction updates following Vlad's comments

1.  moved main.js link to the bottom of html page
2.  corrected comments misspellings 
3.  changed the vue.js plugging from cdn to local and from full-dev to minified.

powpilot committed:
correction updates following Vlad's comments

1.  moved main.js link to the bottom of html page
2.  corrected comments misspellings 
3.  changed the vue.js plugging from cdn to local and from full-dev to minified

powpilot committed:
correction updates following Vlad's comments

1.  moved main.js link to the bottom of html page
2.  corrected comments misspellings 
3.  changed the vue.js plugging from cdn to local and from full-dev to minified.

powpilot committed:
correction updates following Vlad's comments

1.  moved main.js link to the bottom of html page
2.  corrected comments misspellings 
3.  changed the vue.js plugging from cdn to local and from full-dev to minified.

powpilot committed:
correction updates following Vlad's comments

1.  moved main.js link to the bottom of html page
2.  corrected comments misspellings 
3.  changed the vue.js plugging from cdn to local and from full-dev to minified.

powpilot commented:
requested changes have been done
